docs/en/operations/external-authenticators/jwt.md

```diff
@@ -0,0 +1,219 @@
+
+:::note
+Only RS* family algorithms are supported!
+:::
+
+**Example**
+```xml
+<clickhouse>
+    <!- ... -->
+    <jwt_validators>
+        <my_static_jwks_validator>
+          <static_jwks>{"keys": [{"kty": "RSA", "alg": "RS256", "kid": "mykid", "n": "_public_key_mod_", "e": "AQAB"}]}</static_jwks>
+        </my_static_jwks_validator>
+    </jwt_validators>
+</clickhouse>
+```
+
+#### Parameters:
+- `static_jwks` - content of JWKS in json
+- `static_jwks_file` - path to file with JWKS
+
+:::note
+Only one of `static_jwks` or `static_jwks_file` keys must be present in one verifier
+:::
+
+### Verifying JWT signature using JWKS servers {$verifying-jwt-signature-using-static-jwks}
+
+**Example**
+```xml
+<clickhouse>
+    <!- ... -->
+    <jwt_validators>
+        <basic_auth_server>
+          <uri>http://localhost:8000/.well-known/jwks.json</uri>
+          <connection_timeout_ms>1000</connection_timeout_ms>
+          <receive_timeout_ms>1000</receive_timeout_ms>
+          <send_timeout_ms>1000</send_timeout_ms>
+          <max_tries>3</max_tries>
+          <retry_initial_backoff_ms>50</retry_initial_backoff_ms>
+          <retry_max_backoff_ms>1000</retry_max_backoff_ms>
+          <refresh_ms>300000</refresh_ms>
+        </basic_auth_server>
+    </jwt_validators>
+</clickhouse>
+```
+
+#### Parameters:
+
+- `uri` - JWKS endpoint. Mandatory.
+- `refresh_ms` - Period for resend request for refreshing JWKS. Optional, default: 300000.
+
+Timeouts in milliseconds on the socket used for communicating with the server (optional):
+- `connection_timeout_ms` - Default: 1000.
+- `receive_timeout_ms` - Default: 1000.
+- `send_timeout_ms` - Default: 1000.
+
+Retry parameters (optional):
+- `max_tries` - The maximum number of attempts to make an authentication request. Default: 3.
+- `retry_initial_backoff_ms` - The backoff initial interval on retry. Default: 50.
+- `retry_max_backoff_ms` - The maximum backoff interval. Default: 1000.
+
+### Verifying access tokens {$verifying-access-tokens}
+
+Access tokens that are not JWT (and thus no data can be extracted from the token directly) need to be resolved by external providers.
+
+**Example**
+```xml
+<clickhouse>
+    <!- ... -->
+    <access_token_processors>
+        <my_access_token_processor>
+          <provider>google</provider>
+        </my_access_token_processor>
+    </access_token_processors>
+</clickhouse>
+```
+
+#### Parameters:
+
+- `provider` - name of provider that will be used for token processing. Mandatory parameter. Possible options: `google`.
+
+
+### Enabling JWT authentication in `users.xml` {#enabling-jwt-auth-in-users-xml}
+
+In order to enable JWT authentication for the user, specify `jwt` section instead of `password` or other similar sections in the user definition.
+
+Parameters:
+- `claims` - An optional string containing a json object that should be contained in the token payload.
```

Comment:
If take me as reference customer and document reader, it's not clean, what exactly can be in payload.
I found docs for Google (https://developers.google.com/wallet/reference/rest/v1/Jwt) and MS (https://learn.microsoft.com/en-us/entra/identity-platform/access-token-claims-reference) and saw there other fields, but no `resource_access`.
May be add some links here, or samples for different providers?

Reply:
Actually, I inherited this thing, and I think I will remove it: it can be confusing indeed, and its use case is not obvious. Also, it makes code somewhat more complex